MSAL/src/native_auth/network/responses/validator/sign_up/MSALNativeAuthSignUpResponseValidator.swift

```diff
@@ -158,8 +159,8 @@ final class MSALNativeAuthSignUpResponseValidator: MSALNativeAuthSignUpResponseV
             MSALLogger.log(level: .error, context: context, format: "signup/challenge: Unable to decode error response: \(error)")
             return .unexpectedError(.init(errorDescription: MSALNativeAuthErrorMessage.unexpectedResponseBody))
         }
-        if apiError.error == .none {
-            return .unexpectedError(.init(errorDescription: apiError.errorDescription))
+        if apiError.error == .unknownCase {
```

Comment:
We should have the same returned error, the one that contains the errorDescription

Reply:
I changed it to return the full ApiError (errorDescription included) that comes from backend. Before it was creating a new ApiError from scratch using the info of the ApiError from the backend.

If we add new fields in the future we might lose that information in cases like this.

Reply:
Ah, ok, so it is returned, I wasn't sure here so figured I should highlight.